cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/DefaultFlowConfigurer.java

```diff
@@ -80,7 +79,10 @@ public void addFlowlet(String name, Flowlet flowlet) {
   }
 
   private String getFlowletName(Flowlet flowlet) {
-    FlowletDefinition flowletDef = new FlowletDefinition(null, flowlet, 1);
+    DefaultFlowletConfigurer flowletConfigurer = new DefaultFlowletConfigurer(flowlet);
+    flowlet.configure(flowletConfigurer);
+    FlowletSpecification flowletSpecification = flowletConfigurer.createSpecification();
+    FlowletDefinition flowletDef = new FlowletDefinition(name, flowlet, flowletSpecification, 1);
```

Comment:
this seems a roundabout way to get flowlet name, but i didn't want to have code duplication, so using this path to get the flow name.